Sources/SWBBuildSystem/BuildManager.swift

```diff
@@ -59,9 +59,22 @@ package actor BuildManager {
 
         let buildOutputMap: [String:String]?
         if let buildOnlyThesePaths {
-            precondition(request.buildTargets.count == 1, "the client should ensure we have one target only")
-            let buildOnlyTheseOutputs = buildRequestContext.computeOutputPaths(for: buildOnlyThesePaths, workspace: workspaceContext.workspace, target: request.buildTargets.first!, command: request.buildCommand, parameters: request.parameters)
-            buildOutputMap = Dictionary(uniqueKeysWithValues: zip(buildOnlyTheseOutputs, buildOnlyThesePaths.map{ $0.str }))
```

Comment:
This usage of zip could lead to silent data loss because there was no guarantee the arrays were of the same length.